Packages/com.unity.inputsystem/InputSystem/Plugins/EnhancedTouch/EnhancedTouchSupport.cs

```diff
@@ -0,0 +1,120 @@
+using UnityEngine.InputSystem.Utilities;
+
+////TODO: gesture support
+////TODO: mouse/touch simulation support
+////TODO: high-frequency touch support
+
+////REVIEW: have TouchTap, TouchSwipe, etc. wrapper MonoBehaviours like LeanTouch?
+
+namespace UnityEngine.InputSystem.Touch
+{
+    [AddComponentMenu("Input/Enhanced Touch Support")]
+    public class EnhancedTouchSupport : MonoBehaviour
+    {
+        public new static bool enabled => s_Enabled > 0;
+
+        private static int s_Enabled;
+        private static InputSettings.UpdateMode s_UpdateMode;
+
+        private void OnEnable()
+        {
+            Enable();
+        }
+
+        private void OnDisable()
+        {
+            Disable();
+        }
+
+        public static void Enable()
+        {
+            ++s_Enabled;
+            if (s_Enabled > 1)
+                return;
+
+            InputSystem.onDeviceChange += OnDeviceChange;
+            InputSystem.onBeforeUpdate += Touch.BeginUpdate;
+            InputSystem.onSettingsChange += OnSettingsChange;
+
+            SetUpState();
+        }
+
+        public static void Disable()
+        {
+            if (!enabled)
```

Comment:
nitpick: I'd rather use `s_Enabled > 0` then overriding the `enabled` property, which has a defined use for MonoBehaviours - I found it confusing to read this, and it's only used in this one place anyways.